File: Solutions/763. Partition Labels/763. Partition Labels.go
func partitionLabels(s string) []int {
    last_idx:=make(map[int]int)
    for i,c:=range(s){
        last_idx[(int)(c-'a')]=i
    }
    var res []int
    st,ed:=0,0
    for i,c:=range(s){
        ed=max(ed,last_idx[(int)(c-'a')])
        if ed==i{
            res=append(res,ed-st+1)
            st=ed+1
        }
    }
    return res
}

func max(i,j int)int{
    if i>j{
        return i
    }
    return j
}

File: Solutions/763. Partition Labels/763. Partition Labels.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
        vector<int> last_idx(26);
        for(int i=0;i<s.size();++i){
            last_idx[s[i]-'a']=i;
        }
        int ed=0,st=0;
        vector<int> res;
        for(int i=0;i<s.size();++i){
            ed=max(ed,last_idx[s[i]-'a']);
            if(ed==i){
                res.push_back(ed-st+1);
                st=ed+1;
            }
        }
        return res;
    }
};